### PR TITLE
Add a modern accordion element

### PR DESCRIPTION
**Pull request by leofeyer**
This PR adds a modern accordion element now that we have nested fragments. 🥳

### CODE REVIEW

File: core-bundle/contao/templates/twig/content_element/accordion_wrapper.html.twig [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{% extends "@Contao/content_element/_base.html.twig" %}
```
Comment (ausi) [inline core-bundle/contao/templates/twig/content_element/accordion_wrapper.html.twig]: I think we could extend the `accordion.html.twig` template here, create a block `accordion_content` there and only overwrite that part. WDYT?

Comment (leofeyer) [reply core-bundle/contao/templates/twig/content_element/accordion_wrapper.html.twig]: I tried that but the `parent()` call inside the `content` block does not allow adding another block.

Comment (ausi) [reply core-bundle/contao/templates/twig/content_element/accordion_wrapper.html.twig]: Then we could still extend from `accordion.html.twig` and overwrite the whole `content` block?
Because currently the whole script and style tags would be missing here, right?

Comment (leofeyer) [reply core-bundle/contao/templates/twig/content_element/accordion_wrapper.html.twig]: That is true, however, it does not seem to work. 🤔 



Maybe we shouldn‘t extend the `text.html.twig` template?

Comment (leofeyer) [reply core-bundle/contao/templates/twig/content_element/accordion_wrapper.html.twig]: I‘m going to restructure the templates and maybe add a component.

Comment (leofeyer) [reply core-bundle/contao/templates/twig/content_element/accordion_wrapper.html.twig]: Fixed and tested in d73e5794f194bfe46d0e2d11706f9e0d9567ec27.

File: core-bundle/contao/templates/twig/content_element/text.html.twig [hunk 2]
```diff
@@ -1,29 +1,10 @@
...
-{% set content_layout_attributes = attrs(content_layout_attributes|default)
-    .addClass('media media--' ~ layout, layout and image)
-%}
-
-{# By default we add the layout attributes to the wrapper element. If you want to
- disable this behavior, set "default_content_layout" to false. #}
 {% set attributes = attrs(attributes|default)
-    .mergeWith(content_layout_attributes, default_content_layout is not defined or default_content_layout)
+    .addClass('media media--' ~ layout, layout and image)
 %}
```
Comment (ausi) [inline core-bundle/contao/templates/twig/content_element/text.html.twig]: I think this needs to be reverted as this was an explicit extension point for extending templates.

It does not seem to be related to other changes that move the code to the *_text.html.twig* component, right?

Comment (leofeyer) [reply core-bundle/contao/templates/twig/content_element/text.html.twig]: This seems pretty over-engineered to me, especially as the template now only consists of `{{ block('text_component') }}`. You can (and should) now use the `_text.html.twig` component in your extension instead.

Comment (ausi) [reply core-bundle/contao/templates/twig/content_element/text.html.twig]: I think extending the `text.html.twig` template is still a very valid use case, especially on the project level (e.g. for overwriting the default `text.html.twig` template). I don’t see any benefit of this BC break, and I think it is unrelated to the accordion element.

**Review by ausi — DISMISSED** [review]
I’m not a fan of the BC break in https://github.com/contao/contao/pull/6633#discussion_r1431553246

The rest looks very goot to me 🎉

File: core-bundle/contao/templates/twig/content_element/text.html.twig (hunk 2 above)
Comment (m-vo) [reply core-bundle/contao/templates/twig/content_element/text.html.twig]: The idea of the attributes is, that you can provide/define them without overwriting blocks. This is especially helpful for extensions that want to add attributes to logical places but don't care about the specific markup. Not saying that this *must* be like it was or that things are consistent everywhere, but removing a variable, that was intentionally merged (`content_layout_attributes|default`) with one defined in the outer scope by someone else, is potentially breaking stuff.

(Just a quick explanation, I unfortunately didn't have time to review this and the other Twig related PRs properly, yet… :upside_down_face:)

File: core-bundle/contao/templates/twig/content_element/accordion.html.twig [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                        Object.assign({ headerElements, contentElements }, options || {})
+                    );
+
+                    // Show the first element
+                    accordion.folds[0].open();
+                }
+
+                // Change this selector according to your template, or duplicate
+                // the line if you use multiple accordion element classes.
+                initAccordion('.content-accordion > .{{ toggler }}', {
```
Comment (m-vo) [inline core-bundle/contao/templates/twig/content_element/accordion.html.twig:58]: If this is adjustable, we should probably introduce a block. Wdyt?

Comment (leofeyer) [reply core-bundle/contao/templates/twig/content_element/accordion.html.twig:58]: Do you really mean line 60 or rather line 62 (the custom options)?

File: core-bundle/contao/templates/twig/content_element/text.html.twig (hunk 2 above)
Comment (leofeyer) [reply core-bundle/contao/templates/twig/content_element/text.html.twig]: Reverted in 2b028470e5bec31618e9af80ccc480c025488e14.

File: core-bundle/contao/templates/twig/content_element/accordion.html.twig (hunk 3 above)
Comment (m-vo) [reply core-bundle/contao/templates/twig/content_element/accordion.html.twig:58]: I'm not sure. The comment suggests that the selector should be adjusted if needed. How can I do this without duplicating the block?

Comment (m-vo) [reply core-bundle/contao/templates/twig/content_element/accordion.html.twig:58]: But probably also the options, yes. 🙂

Comment (leofeyer) [reply core-bundle/contao/templates/twig/content_element/accordion.html.twig:58]: The comment is a leftover from the `js_accordion` template, which we do not need here, because the toggler class is already customizable in the module settings. But a custom block for the options sure makes sense. 👍

Comment (leofeyer) [reply core-bundle/contao/templates/twig/content_element/accordion.html.twig:58]: Added in ca8c62d9f475e712d94e841f84d7b08603ffab8d.

**Comment by leofeyer** [comment]
Better approach is here: #6648
